File: backend/src/main/java/com/collegeevent/service/VolunteerService.java
package com.collegeevent.service;

import com.collegeevent.entity.Volunteer;

import java.util.List;
import java.util.Optional;

public interface VolunteerService {

    Volunteer saveVolunteer(Volunteer volunteer);

    List<Volunteer> getAllVolunteers();

    Optional<Volunteer> getVolunteerById(Long id);

    Volunteer updateVolunteer(Long id, Volunteer volunteer);

    void deleteVolunteer(Long id);
}

File: frontend/src/services/volunteerService.js
import api from "./api";

export const getAllVolunteers = () => {
    return api.get("/volunteers");
};

export const getVolunteerById = (id) => {
    return api.get(`/volunteers/${id}`);
};

export const applyVolunteer = (volunteer) => {
    return api.post("/volunteers", volunteer);
};

export const updateVolunteer = (id, volunteer) => {
    return api.put(`/volunteers/${id}`, volunteer);
};

export const deleteVolunteer = (id) => {
    return api.delete(`/volunteers/${id}`);
};